lib/vagrant/ui.rb

```diff
@@ -252,6 +265,19 @@ def format_message(type, message, **opts)
       end
     end
 
+
+    class NonInteractive < Basic
+      def initialize
+        super
+        @opts = { :show_progress => false }
+      end
+
+      def ask(*args)
+        # Non interactive can't ask for input
+        raise Errors::UIExpectsTTY
+      end
```

Comment:
Should include a no-op `rewriting` method. 

The `clear_line` method should probably also be overridden. I think the correct behavior would be throwing an error when used, but that would likely introduce unexpected breakages simply for correctness. Might be better to just emit a warning for now, and output a newline instead?

Reply:
Sounds good. 
When outputting a new line the result of `vagrant up box` is 
```
...
==> vbox: Loading metadata for box 'bento/ubuntu-18.04'
    vbox: URL: https://vagrantcloud.com/bento/ubuntu-18.04
==> vbox: Adding box 'bento/ubuntu-18.04' (v202002.14.0) for provider: virtualbox
    vbox: Downloading: https://vagrantcloud.com/bento/boxes/ubuntu-18.04/versions/202002.14.0/providers/virtualbox.box

    vbox: Download redirected to host: vagrantcloud-files-production.s3.amazonaws.com
...
```


When not outputting a new line the result of `vagrant up box` is 
```
...
==> vbox: Loading metadata for box 'bento/ubuntu-18.04'
    vbox: URL: https://vagrantcloud.com/bento/ubuntu-18.04
==> vbox: Adding box 'bento/ubuntu-18.04' (v202002.14.0) for provider: virtualbox
    vbox: Downloading: https://vagrantcloud.com/bento/boxes/ubuntu-18.04/versions/202002.14.0/providers/virtualbox.box
==> vbox: Box download is resuming from prior download progress
    vbox: Download redirected to host: vagrantcloud-files-production.s3.amazonaws.com
...
```

I think outputting a new line looks a little more messy. But I'm happy to keep it in if you think it's useful for users to see that

Reply:
I agree, it does look messy, but I think that might be the result of `clear_line` not being used correctly in that location. Generally, when it is used, a line is written without a new line attached so the cursor stays on the line. The clear line will remove the content and return the cursor to the start of the line and then write again. If progress output, or some other type of dynamically modified output, is being displayed and is not wrapped with the `rewriting` method, the result will be one extremely long line. So I think the behavior of writing a new line is okay here, but we'll need to cleanup the clear_line usage internally.